StreamingClient.Base/Web/LocalHttpListenerServer.cs

```diff
@@ -127,7 +127,10 @@ protected string GetRequestParameter(HttpListenerContext listenerContext, string
                     {
                         token = token.Substring(0, endIndex);
                     }
-                    return token;
+
+                    String decodedToken = HttpUtility.UrlDecode(token);
```

Comment:
This code is used across all of the supported platforms, so this shouldn't be changed here if there's a specific case in YouTube. We don't want unnecessary decoding to adversely effect the value if it's not intended to be decoded.

Are there specific areas that you have identified where URL decoding is necessary for YouTube interactions and if so what are they? For those cases, we can add an optional parameter to the function that indicates whether the value should be URL decoded or not.

Reply:
I suppose it isn't only Youtube. The OAuth specs indicate that url queries 'may' be url encoded using 'application/x-www-form-urlencoded':
https://tools.ietf.org/html/rfc6749#section-3.1
https://tools.ietf.org/html/rfc6749#section-3.1.2

Reply:
The fun thing about spec docs is always hoping that people who build systems from them actually follow the docs. I've seen my fair share "interesting" designs for OAuth services that I've worked with. 😄 

Do you specifically know if there's any requests from YouTube that do require encoding? Or was this more a precautionary thing? Not against the idea of adding it, I just want to make sure there wouldn't be any adverse effects to other services.

Typically any API services that do use encoding will mention it somewhere in their documentation (or not and you play the guessing game). Some of the OAuth services I know that do require URL encoding are Discord, JustGiving, Patreon, & StreamElements. However a fair amount I know don't do any sort of encoding. The good ones will usually make a note of it somewhere in their doc (EX: https://discord.com/developers/docs/topics/oauth2)

Reply:
The authentication codes I receive from Youtube often contain a / which is encoded in the url as %2F. For example when the redirect url is used with the build-in local HTTP listener it redirects to url: http://localhost:8919/?code=4%2F1AHq_... The current implementation sends back the 4%2FAHq... code to Youtube which fails. When decoded as 4/1AHq... it succeeds.

The same applies to all query parameters received in the redirect url, for example scope:
"&scope=email+profile+https%3A%2F%2Fwww.googleapis.com"

Reply:
Ok then let's go off that assumption and implement it for YouTube only for right now. So if we add an optional bool to the function to decode the parameter, we can then just set that for all of the YouTube interactions.